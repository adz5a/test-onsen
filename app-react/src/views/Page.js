import React, {  } from 'react';
import {
    Page as OnsenPage,
    Toolbar as OnsenToolbar,
    BackButton,
    Icon
} from "react-onsenui";


export function goBack () {

    history.back(); // eslint-disable-line no-restricted-globals

}



export function createToolbar ( {
    title,
    hasBackButton
} ) {

    return () => (
        <OnsenToolbar>
            <div className="left">
                {

                    hasBackButton ?
                        <BackButton onClick={goBack}>Back</BackButton>:
                        null


                }
            </div>
            <div className="center">{title}</div>
            <div className="right">
                <Icon icon="md-menu"/>
            </div>
        </OnsenToolbar>
    );

}


export function Page ( {
    title = "",
    hasBackButton = false,
    children
} ) {

    return (
            <OnsenPage
                renderToolbar={createToolbar({ title, hasBackButton })}
            >
                { children }
            </OnsenPage>
    );

}
