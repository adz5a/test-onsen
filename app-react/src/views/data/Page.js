import React, {
    // Component,
    // PropTypes
} from "react";
import {
    Page
} from "./../Page";
import {
    compose,
    lifecycle
} from "recompose";
import {
    LOAD
} from "data/aws";
import {
    connect
} from "react-redux";
import {
    Route,
    Switch
} from "react-router-dom";
import {Card, CardActions, CardHeader,} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import TextField from 'material-ui/TextField';

const cardStyle = {
    maxWidth: "30em",
    margin: "auto"
};
const pStyle = {
    display: "flex",
    justifyContent: "space-around",
    alignItems: "center"
};

const title = "Bucket Information";


function BucketForm () {

    return (
        <Card
            style={cardStyle}
        >
            <CardHeader
                title={title}
            />
            <CardActions>
                <form
                    onSubmit={ e => {

                        e.preventDefault();
                        const form = e.target.elements;
                        const data = {
                            baseURL: form.baseURL.value,
                            bucket: form.bucket.value
                        };

                        console.log(data);
                    }}
                >
                    <div style={pStyle}>
                        <span>Base URL</span><TextField name="baseURL" />
                    </div>
                    <div style={pStyle}>
                        <span>Bucket Name</span><TextField name="bucket" />
                    </div>
                    <FlatButton type="submit">Ok</FlatButton>
                </form>
            </CardActions>
        </Card>
    );

}


export function DataPage () {

    return (
        <Page
            title="Data"
            hasBackButton={true}
        >
            <section
                style={{
                    paddingTop: "3em",
                    height: "100%"
                }}
            >
                <BucketForm />
            </section>
        </Page>

    )

}
