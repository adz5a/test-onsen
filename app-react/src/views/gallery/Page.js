import React, {
    // Component,
    // PropTypes
}Â from "react";
import {
    Page
} from "./../Page";
import {
} from "recompose";

export function GalleryPage () {

    return (
        <Page
            title="Gallery"
            hasBackButton={true}
        >
            <section
                style={{
                    maxWidth: "80%",
                    margin: "auto",
                    paddingTop: "3em",
                    height: "100%"
                }}
            >
            </section>
        </Page>

    )

}
