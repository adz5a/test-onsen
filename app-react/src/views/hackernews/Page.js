import React, {
    // Component,
    // PropTypes
}Â from "react";
import {
    Page
} from "./../Page";
import {
} from "recompose";
import {
    Card,
    CardHeader,
    CardActions
} from "material-ui/Card";
import {
    Link
} from "react-router-dom";
import FlatButton from 'material-ui/FlatButton';

export function HackernewsPage () {

    return (
        <Page
            title="Not Found"
            hasBackButton={true}
        >
            <section
                style={{
                    maxWidth: "80%",
                    margin: "auto",
                    paddingTop: "3em",
                    height: "100%"
                }}
            >
                yolo
            </section>
        </Page>

    )

}

