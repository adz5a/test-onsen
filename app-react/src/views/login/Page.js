import React, { 
    // Component,
    // PropTypes
}Â from "react";
import {
    Page
} from "./../Page";
import {
    ConnectedForm as Form
} from "./Form";


export function LoginPage () {

    return (
        <Page
            title="Login"
            hasBackButton={true}
        >
            <section
                style={{
                    maxWidth: "80%",
                    margin: "auto",
                    paddingTop: "3em"
                }}
            >
                <Form />
            </section>
        </Page>

    )

}
