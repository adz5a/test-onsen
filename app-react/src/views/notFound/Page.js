import React, {
    // Component,
    // PropTypes
}Â from "react";
import {
    Page
} from "./../Page";
import {
} from "recompose";
export function NotFoundPage () {

    return (
        <Page
            title="Not Found"
            hasBackButton={true}
        >
            <section
                style={{
                    maxWidth: "80%",
                    margin: "auto",
                    paddingTop: "3em",
                    height: "100%"
                }}
            >
                <p>
                    {"Not FOUND"}
                </p>
            </section>
        </Page>

    )

}
