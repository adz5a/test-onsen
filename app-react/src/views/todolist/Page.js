import React, {
    // Component,
    // PropTypes
} from "react";
import {
    Page
} from "./../Page";
import {
    ConnectedForm as Form
} from "./Form";
import {
    ConnectedList as List
}from "./List";
import {
    isLogged
} from "data/user";
import {
    STATUS_ON,
    STATUS_OFF
} from "data/todolist";
import {
    Unauthorized
} from "./Unauthorized";
import {
    branch,
    compose,
    renderComponent
} from "recompose";
import {
    connect
} from "react-redux";
import {
    Tabs,
    Tab
} from "material-ui/Tabs";

const todoListStyle = {
    paddingLeft: "5em",
    paddingRight: "5em"
};
function TodoList () {

    return (
        <section>
            <Form />
            <section
                style={todoListStyle}
            >
                <Tabs>
                    <Tab
                        label="All"
                    >
                        <List
                        />
                    </Tab>
                    <Tab
                        label="Actives"
                    >
                        <List
                            status={STATUS_ON}
                        />
                    </Tab>
                    <Tab
                        label="Inactives"
                    >
                        <List
                            status={STATUS_OFF}
                        />
                    </Tab>
                </Tabs>
            </section>
        </section>
    );

}

const WithUser = connect(
    state => ({ user: state.user })
);

const WhenLogged = branch(
    ( { user } ) => isLogged(user),
    renderComponent(TodoList),
);

const Content = compose(
    WithUser,
    WhenLogged
)(Unauthorized);

export function TodoListPage () {

    return (
        <Page
            title="Todo List"
            hasBackButton={true}
        >
            <section>
                <Content />
            </section>
        </Page>
    );

}
