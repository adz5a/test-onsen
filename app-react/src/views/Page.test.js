import React from 'react';
import ReactDOM from 'react-dom';
import reactTapEventPlugin from "react-tap-event-plugin";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import { Page } from "./Page";
import renderer from "react-test-renderer";


// necessary for the onTap prop
reactTapEventPlugin();

function createNodeMock ( name ) {

    return {

        style: {}

    };

}


it('renders without crashing', () => {
    ReactDOM.render(
        <MuiThemeProvider>
            <Page />
        </MuiThemeProvider>,
        document.createElement("div")
    );
});

it("matches snapshot", () => {


    const tree = renderer.create(
        <MuiThemeProvider>
            <Page />
        </MuiThemeProvider>,
        {Â createNodeMock }
    ).toJSON();

    expect(tree).toMatchSnapshot();


});
