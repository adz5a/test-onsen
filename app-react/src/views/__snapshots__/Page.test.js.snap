// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#00bcd4",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100,
      }
    }
  >
    <h1
      onTouchTap={[Function]}
      style={
        Object {
          "boxFlex": 1,
          "color": "#ffffff",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      
    </h1>
    <div
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": -16,
          "marginTop": 8,
          "muiPrepared": true,
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fill": "#ffffff",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M4 15h16v-2H4v2zm0 4h16v-2H4v2zm0-8h16V9H4v2zm0-6v2h16V5H4z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <section />
</section>
`;
